reporting.txt

```diff
@@ -0,0 +1,952 @@
+      successfully negotiate a policy-compliant TLS connection, and
+      serves to provide a "heartbeat" to receiving domains that
+      reporting is functional and tabulating correctly.
+
+4.1.2.  Routing Failures
+
+   o  "mx-mismatch": This indicates that the MX resolved for the
+      recipient domain did not match the MX constraint specified in the
+      policy.
+
+   o  "certificate-mismatch": This indicates that the certificate
+      presented by the receiving MX did not match the MX hostname
+
+4.1.3.  Negotiation Failures
+
+   o  "starttls-not-supported": This indicates that the recipient MX did
+      not support STARTTLS.
+
+   o  "invalid-certificate": This indicates that the certificate
+      presented by the receiving MX did not validate according to the
+      policy validation constraint.  (Either it was not signed by a
+      trusted CA or did not match the DANE TLSA record for the recipient
+      MX.)
+
+   o  "expired-certificate": This indicates that the certificate has
+      expired.
+
+4.1.3.1.  DANE-specific Policy Failures
+
+   o  "tlsa-invalid": This indicates a validation error for Policy
+      Domain specifying "tlsa" validation.
+
+   o  "dnssec-invalid": This indicates a failure to validate DNS records
+      for a Policy Domain with a published "tlsa" record.
+
+
+
+
+
+
+
+
+Margolis, et al.        Expires October 20, 2016                [Page 6]
+
+Internet-Draft                 SMTP-TLSRPT                    April 2016
+
+
+4.1.3.2.  STS-specific Policy Failures
+
+   o  "sts-invalid": This indicates a validation error for Policy Domain
+      specifying "STS" validation.
+
+   o  sender-does-not-support-validation-method: This indicates the
+      sending system can never validate using the requested validation
+      mechanism.
+
+5.  Reporting Delegation
+
+   It is possible to specify destinations for the different reports that
+   are outside the authority of the Policy Domain making the request.
+   This allows domains that do not operate mail servers to request
+   reports and have them go someplace that is able to receive and
+   process them.
+
+   Without checks, this would allow a bad actor to publish a TLS-RPT
+   policy record that requests that reports be sent to a victim address,
+   potentially causing the victim to be flooded with unwanted reports.
+   Therefore, a verification mechanism is included as inspired by
+   Section 7.1 from [RFC7489].
+
+   When a sending MTA discovers an TLS-RPT policy in the DNS, and the
+   Organizational Domain at which that record was discovered is not
+   identical to the Organizational Domain of the host part of the
+   authority component of a [URI] specified in the "rua" or "ruf" tag,
+   the following verification steps are to be taken:
+
+   1.  Extract the host portion of the authority component of the URI.
+       Call this the "destination host", as it refers to a Report
+       Receiver.
+
+   2.  Prepend the string "_report._tlsrpt".
```

Comment:
Should this be reversed?  The more generic portion closer to the domain?  Such as currently they do things like _25._tcp.mx1.foo.com for DANE, and tcp is more generic than the port.